iavl_versioned_tree.go

```diff
@@ -0,0 +1,167 @@
+package iavl
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	dbm "github.com/tendermint/tmlibs/db"
+)
+
+var ErrVersionDoesNotExist = fmt.Errorf("version does not exist")
+
+// VersionedTree is a persistent tree which keeps track of versions.
+type VersionedTree struct {
+	*orphaningTree                           // The current, latest version of the tree.
+	versions       map[uint64]*orphaningTree // The previous, saved versions of the tree.
+	latest         uint64                    // The latest saved version.
```

Comment:
maybe call it `latestSaved` than

Reply:
I changed the comment on *orphaningTree to be less confusing and renamed latest to latestVerison. 411c361

Effectively, a version can only exist if it is saved.